tests/stress/km/stress_tests_km.cpp

```diff
@@ -1084,6 +1091,280 @@ _print_test_control_info(const test_control_info& test_control_info)
+    auto [program_object, _] =
+        _load_attach_tail_program(file_name, 0, EBPF_ATTACH_TYPE_BIND, program_name, program_type);
+
+    // Set up the tail call programs.
+    fd_t map_fd = _set_up_tailcall_program(program_object.get(), map_name);
+
+    // Needed for thread_context initialization.
+    constexpr uint32_t MAX_BIND_PROGRAM = 1;
+
+    // Storage for object pointers for each ebpf program file opened by bpf_object__open().
+    std::vector<object_table_entry> object_table(MAX_BIND_PROGRAM);
+    for (uint32_t index = 0; auto& entry : object_table) {
+        entry.available = true;
+        entry.lock = std::make_unique<std::mutex>();
+        entry.object = std::move(program_object);
+        entry.attach = !(index % 2) ? true : false;
+        entry.index = index++;
+        entry.reuse_count = 0;
+        entry.tag = 0xC001DEA2;
+    }
+
+    size_t total_threads = test_control_info.threads_count;
+    std::vector<thread_context> thread_context_table(
+        total_threads, {{}, {}, false, {}, thread_role_type::ROLE_NOT_SET, 0, 0, 0, false, 0, 0, object_table});
+    std::vector<std::thread> test_thread_table(total_threads);
+    for (uint32_t i = 0; i < total_threads; i++) {
+
+        // First, prepare the context for this thread.
+        auto& context_entry = thread_context_table[i];
+        context_entry.is_native_program = true;
+        context_entry.role = thread_role_type::MONITOR_IPV6;
+        context_entry.thread_index = i;
+        context_entry.duration_minutes = test_control_info.duration_minutes;
+        context_entry.extension_restart_enabled = test_control_info.extension_restart_enabled;
+
+        // Now create the thread.
+        auto& thread_entry = test_thread_table[i];
+        thread_entry =
+            std::move(std::thread(_invoke_mt_bindmonitor_tail_call_thread_function, std::ref(context_entry)));
+    }
+
+    // Another table for the 'extension restart' threads.
+    std::vector<std::thread> extension_restart_thread_table{};
+
+    // If requested, start the 'extension stop-and-restart' thread for extension for this program type.
+    std::string extension_name = {"netebpfext"};
+    if (test_control_info.extension_restart_enabled) {
+        auto restart_thread = _start_extension_restart_thread(
+            std::ref(extension_name), test_control_info.extension_restart_delay_ms, test_control_info.duration_minutes);
+        extension_restart_thread_table.push_back(std::move(restart_thread));
+    }
+
+    // Wait for threads to terminate.
+    LOG_INFO("waiting on {} test threads...", test_thread_table.size());
+    for (auto& t : test_thread_table) {
+        t.join();
+    }
+
+    if (test_control_info.extension_restart_enabled) {
+        LOG_INFO("waiting on {} extension restart threads...", extension_restart_thread_table.size());
+        for (auto& t : extension_restart_thread_table) {
+            t.join();
+        }
+    }
+
+    // Clean up tail call programs.
```

Comment:
I'd recommend removing this clean-up step and deliberately let ebpfcore do the required 'post-app-exit' clean-up.  It should be able to do so without issues, especially after this sort of a stress run.

Reply:
Does the post-app-exit clean all the tail calls?

In my understanding, the clean up is required.

Reply:
Removed

Reply:
Yes, the in-kernel post-app-exit handling in ebpf _is_ _required_ _to_ ensure that.  Not doing so is a bug.
(I'd point to the cases where a badly written application either fails to do so, or a well written application terminates/crashes before it can).

In this case, we _want_ that prog array map to contain valid program fds so that when the application (deliberately) terminates without clean-up, epbf is forced to do this internally.